src/components/ec-timer/ec-timer.vue

```diff
@@ -0,0 +1,144 @@
+<template>
+  <div
+    class="ec-timer"
+    data-test="ec-timer"
+  >
+    <svg
+      :width="diameter"
+      :height="diameter"
+      :viewBox="viewbox"
+      class="ec-timer__svg"
+    >
+      <circle
+        class="ec-timer__elapsed"
+        cx="50%"
+        cy="50%"
+        :r="radius"
+        fill="transparent"
+        :stroke-width="strokeWidth"
+      />
+
+      <circle
+        class="ec-timer__remaining"
+        cx="50%"
+        cy="50%"
+        :r="radius"
+        fill="transparent"
+        :stroke-width="strokeWidth"
+        :stroke-dasharray="circumference"
+        :stroke-dashoffset="offset"
+      />
+
+      <text
+        x="50%"
+        y="50%"
+        data-test="ec-timer__text"
+        class="ec-timer__text"
+        dominant-baseline="central"
+        text-anchor="middle"
+      >
+        {{ secondsLeft }}<slot>s</slot>
+      </text>
+    </svg>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'EcTimer',
+  props: {
+    seconds: {
+      type: Number,
+      required: true,
+      validator(value) {
+        return Number.isInteger(value) && value > 0;
+      },
+    },
+    isRunning: {
+      type: Boolean,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      radius: 24,
+      strokeWidth: 4,
+      secondsLeft: this.seconds,
+      startTime: null,
+      currentTime: null,
+    };
+  },
+  computed: {
+    diameter() {
+      return (this.radius * 2) + this.strokeWidth;
+    },
+    viewbox() {
+      return `0 0 ${this.diameter} ${this.diameter}`;
+    },
+    circumference() {
+      return 2 * Math.PI * this.radius;
+    },
+    steps() {
+      return this.circumference / this.seconds;
+    },
+    offset() {
+      return this.circumference + this.steps * this.secondsLeft;
+    },
+    timeDifference() {
+      return this.currentTime - this.startTime;
+    },
+  },
+  watch: {
+    isRunning: {
+      immediate: true,
+      handler(value) {
+        if (value) {
+          this.countdown();
+        } else {
+          clearInterval(this.timerInterval);
+          this.secondsLeft = this.seconds;
+        }
+      },
+    },
+  },
+  beforeDestroy() {
+    clearInterval(this.timerInterval);
+  },
+  methods: {
+    countdown() {
+      this.startTime = Math.ceil(Date.now() / 1000);
+      this.timerInterval = setInterval(this.reduceSecondsLeft, 1000);
+    },
+    reduceSecondsLeft() {
+      this.currentTime = Math.ceil(Date.now() / 1000);
+      if (this.timeDifference < this.seconds) {
+        this.secondsLeft = this.seconds - this.timeDifference;
+      } else {
+        this.secondsLeft = this.seconds - this.timeDifference;
+        clearInterval(this.timerInterval);
+        this.$emit('time-expired');
+      }
+    },
```

Comment:
Since we are repeating the same line in both if-blocks, could we move it out? Something like:
```suggestion
    reduceSecondsLeft() {
      this.currentTime = Math.ceil(Date.now() / 1000);
      this.secondsLeft = this.seconds - this.timeDifference;
      if (this.timeDifference >= this.seconds) {
        clearInterval(this.timerInterval);
        this.$emit('time-expired');
      }
    },
```

Reply:
You are absolutely right